modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/describer/AnnotationsBasedDescriber.java

```diff
@@ -364,7 +367,11 @@ private void declareMessageSource(HasSourceDeclarer declarer, Class<? extends So
 
         sourceDeclarers.put(sourceType, source);
         declareMetadataKeyId(sourceType, source);
-        declareSingleParameters(getParameterFields(sourceType), source, MuleExtensionAnnotationParser::parseMetadataAnnotations);
+        declareSingleParameters(getParameterFields(sourceType)
+                                        .stream()
+                                        .filter(field -> !isMultiLevelMetadataKeyId(field, field.getType(), typeLoader))
```

Comment:
I don't think this is the correct place to do this. The declareSingleParameters method should know how to treat fields with this characteristics. Otherwise the feature is not self contained and you have bits and pieces of the solution scattered around